rbac/common/base/base_message.py

```diff
@@ -225,28 +362,187 @@ def get(self, object_id, target_id=None):
             target_id=target_id,
         )
 
+    def get_addresses(self, context, addresses):
+        """Get the deserialized state entries given a list of blockchain addresses"""
+        state_entries = {}
+        records = state_client.get_addresses(context=context, addresses=addresses)
+        if not records:
+            return state_entries
+        for record in records:
+            state_entries[record.address] = addresser.deserialize(
+                address=record.address, data=record.data
+            )
+        return state_entries
+
+    def _make_output_state(self, input_state, outputs):
+        """Makes the output state entries from the input state and
+        adds the addresses that were not in state"""
+        output_state = {"changed": set()}
+        inputs_to_outputs = [address for address in outputs if address in input_state]
+        for address in inputs_to_outputs:
+            output_state[address] = input_state[address]
+        return output_state
+
     def message_to_storage(self, message):
         """Transforms the message into the state (storage) object"""
-        # pylint: disable=not-callable
         return batcher.message_to_message(
-            self._state_object(), self._name_camel, message
+            # pylint: disable=not-callable
+            message_to=self._state_object(),
+            message_from=message,
+            message_name=self._name_camel,
+            exclude_fields=self.message_fields_not_in_state,
         )
 
-    def set_state(self, state, message, object_id, target_id=None):
+    def set_state(self, context, message, outputs, object_id, target_id=None):
         """Creates a new address in the blockchain state"""
         store = self.message_to_storage(message=message)
         # pylint: disable=no-member,not-callable
         container = self._state_container()
-        container.users.extend([store])
+        getattr(container, self._state_container_list_name).extend([store])
+        address = self.address(object_id=object_id, target_id=target_id)
+        if address not in outputs:
+            raise ValueError(
+                "Address {} not in listed outputs".format(addresser.parse(address))
+            )
+        state_client.set_address(context=context, address=address, container=container)
+
+    def _get_store(self, object_id, target_id, outputs, output_state):
+        """Gets the store object to store data for this message"""
+        address = self.address(object_id=object_id, target_id=target_id)
+        container = None
+        if address not in outputs and address in output_state:
+            raise ValueError(
+                "Address {} not in listed outputs".format(addresser.parse(address))
+            )
+        if address in output_state:
+            container = output_state[address]
+            # TODO: is getting the first item in the container... may not be correct!
```

Comment:
Have you made a ticket for this?  Or written tests to check if it gets the first item?

Reply:
We have no present functionality that writes multiple records into a container yet. This will probably be correct if we add records to the top, but I wanted to leave the note there to make sure we are aware of the issue.